lookup_plugin/s3_handler.py

```diff
@@ -22,12 +28,200 @@ def load_parameters_from_json(filename):
+            break
+
+def pause_gcProcess(pid):
+    try:
+        pid = int(pid)  # Convert string pid to integer
+    except ValueError:
+        logging.error(f"pause_gc_process: Invalid PID format: {pid}")
+        return -1
+
+    try:
+        process_obj = psutil.Process(pid)
+    except psutil.NoSuchProcess:
+        logging.error(f"pause_process: Process with PID {pid} not found")
+        return -1
+
+    logging.info(f"Pausing process {pid} by sending SIGSTOP")
+    try:
+        process_obj.send_signal(signal.SIGSTOP)
+        return 0
+    except PermissionError as e:
+        logging.error(f"resume_process: Insufficient permissions to send signal: {e}")
+        return -1
+    except psutil.NoSuchProcess:  # Handle cases where process terminates during resume
+        logging.error(f"resume_process: Process {pid} terminated unexpectedly")
+        return -1
+    except Exception as e:  # Catch unexpected errors
+        logging.error(f"resume_process: Unexpected error: {e}")
+        return -1
+
+def resume_gcProcess(pid):
+    try:
+        pid = int(pid)  # Convert string pid to integer
+    except ValueError:
+        logging.error(f"pause_gc_process: Invalid PID format: {pid}")
+        return -1
+
+    try:
+        process = psutil.Process(pid)
+    except psutil.NoSuchProcess:
+        logging.error(f"pause_process: Process with PID {pid} not found")
+        return -1
+
+    logging.info(f"Resuming process {pid} by sending SIGCONT")
+
+    try:
+        process.send_signal(signal.SIGCONT)
+        return 0
+    except PermissionError as e:
+        logging.error(f"resume_process: Insufficient permissions to send signal: {e}")
+        return -1
+    except psutil.NoSuchProcess:  # Handle cases where process terminates during resume
+        logging.error(f"resume_process: Process {pid} terminated unexpectedly")
+        return -1
+    except Exception as e:  # Catch unexpected errors
+        logging.error(f"resume_process: Unexpected error: {e}")
+        return -1
+
+def run_dummyData_cmd(command):
+    print("command", command)
+    try:
+        result = subprocess.run(command, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Error: {e}")
+
+def generate_dbi_dbo_concurrently(cluster_params, dirName):
+    base_dir = cluster_params['base_dir']
+    raft_uuid = cluster_params['raft_uuid']
+    s3Support = cluster_params['s3Support']
+    binary_dir = os.getenv('NIOVA_BIN_PATH')
+    bin_path = '%s/dummyData' % binary_dir
+    path = "%s/%s/%s/" % (base_dir, raft_uuid, dirName)
+    if not os.path.exists(path):
+        # Create the directory path
+        try:
+            os.makedirs(path, mode=0o777)
+        except Exception as e:
+            print(f"An error occurred while creating '{path}': {e}")
+
+    s3configPath = '%s/s3.config.example' % binary_dir
+    s3LogFile = "%s/%s/s3Upload" % (base_dir, raft_uuid)
+
+    commands = []
+    for chunk in range(1, 13):
```

Comment:
this will be used for only 12 chunks,  by passing as an argument you can reuse this.